File: apps/auth-server/src/main/java/com/example/chat/auth/server/infrastructure/persistence/JpaCredentialRepository.java
package com.example.chat.auth.server.infrastructure.persistence;

import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;
import java.util.UUID;

public interface JpaCredentialRepository extends JpaRepository<CredentialEntity, UUID> {
    Optional<CredentialEntity> findByPrincipalIdAndType(UUID principalId, String type);

    void deleteByPrincipalIdAndType(UUID principalId, String type);
}

File: apps/auth-server/src/main/kotlin/com/example/chat/auth/server/infrastructure/persistence/JpaCredentialRepository.kt
package com.example.chat.auth.server.infrastructure.persistence

import java.util.*
import org.springframework.data.jpa.repository.JpaRepository

interface JpaCredentialRepository : JpaRepository<CredentialEntity, UUID> {
    fun findByPrincipalIdAndType(principalId: UUID, type: String): Optional<CredentialEntity>
    fun deleteByPrincipalIdAndType(principalId: UUID, type: String)
}
